android/modules/calendar/src/java/ti/modules/titanium/calendar/RecurrenceRuleProxy.java

```
@@ -0,0 +1,376 @@
+			} catch (ParseException e) {
+				e.printStackTrace();
+			}
+		}
+
+		// Check for repeat count.
+		String until = matchExpression(".*(COUNT=[0-9]*).*", 6);
+		if (until != null) {
+			this.endDictionary.put(this.count, until);
+		}
+	}
+
+	private void calculateInterval() {
+		String interval = matchExpression(".*(INTERVAL=[0-9]*).*", 9);
+		if (interval != null) {
+			this.interval = Integer.valueOf(interval);
+		}
+	}
+	//endregion
+
+	//region kroll proxy methods
+	@Kroll.getProperty @Kroll.method
+	public String getCalendarID() {
+		return this.calendarID;
+	}
+
+	@Kroll.getProperty @Kroll.method
+	public int[] getDaysOfTheMonth() {
+		return this.daysOfTheMonth;
+	}
+
+	@Kroll.getProperty @Kroll.method
+	public KrollDict[] getDaysOfTheWeek() {
+		return this.daysOfTheWeek;
+	}
+
+	@Kroll.getProperty @Kroll.method
+	public int[] getDaysOfTheYear() {
+		return this.daysOfTheYear;
+	}
+
+	@Kroll.getProperty @Kroll.method
+	public KrollDict getEnd() {
+		return this.endDictionary;
+	}
+
+	@Kroll.getProperty @Kroll.method
+	public int getFrequency() {
+		return this.frequency.toTiIntId();
+	}
+
+	@Kroll.getProperty @Kroll.method
+	public int getInterval() {
+		return this.interval;
+	}
+
+	@Kroll.getProperty @Kroll.method
+	public int[] monthsOfTheYear() {
+		return this.monthsOfTheYear;
+	}
+
+	@Kroll.getProperty @Kroll.method
+	public int[] getWeeksOfTheYear() {
+		return this.weeksOfTheYear;
+	}
+	//endregion
+
+
+	private void parseRecurrenceRule(String rrule, int calendarID) {
+		if (rrule == null) {
+			return;
+		}
+		// Set the recurrence rule for future use.
+		this.rRule = rrule;
+
+		// Set the ID of the calendar.
+		this.calendarID = String.valueOf(calendarID);
+
+		// Set the frequency in constructor, because it is required for other properties.
+		String frequency = matchExpression(".*(FREQ=[A-Z]*).*", 5);
+		if (frequency != null) {
+			this.frequency = TiRecurrenceFrequencyType.fromRfcStringId(frequency);
```

The static method can return null. Should we throw an exception or default to DAILY?